Changed code to improve output for test/e2e/windows

NikhilSharmaWe opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
Enhancement
<!--
Add one of the following kinds:
/kind cleanup

Optionally add one or more of the following kinds if applicable:
-->

#### What this PR does / why we need it:
For better (more informative) output for developers when test fails. Changed test/e2e/windows for this PR.
#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Part of #105678

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

NikhilSharmaWe commented:
@pohly could you please review the changes made

NikhilSharmaWe commented:
@pohly could you please review the changes made

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
 				}
-				framework.ExpectEqual(found, true, "Found pod in stats summary")
+				if !found{
+					framework.Failf("Pod with name : %v and namespace : %v not found in the stats summary %+v", p.Name, p.Namespace, allPods)
```
I'm not a fan of ` : ` as separator between name/value pairs. Also name/namespace pairs usually just get printed as `<namespace>/<name>`.

```suggestion
					framework.Failf("Pod with name : %v and namespace : %v not found in the stats summary %+v", p.Name, p.Namespace, allPods)
```
```suggestion
					framework.Failf("Pod %s/%s not found in the stats summary %+v", p.Namespace, p.Name, allPods)
```

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
 				framework.Logf("Pod %s usage: %v", p.Name, cpuUsage)
-				framework.ExpectEqual(cpuUsage > 0, true, "Pods reported usage should be > 0")
-				framework.ExpectEqual((.5*1.05) > cpuUsage, true, "Pods reported usage should not exceed limit by >5%")
+				if !(cpuUsage > 0){
```
```suggestion
				if cpuUsage <= 0 {
```

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
-				framework.ExpectEqual(cpuUsage > 0, true, "Pods reported usage should be > 0")
-				framework.ExpectEqual((.5*1.05) > cpuUsage, true, "Pods reported usage should not exceed limit by >5%")
+				if !(cpuUsage > 0){
+					framework.Failf("Pods reported usage is %v, but it should be greater than 0", cpuUsage)
```
```suggestion
					framework.Failf("Pod %s/%s reported usage is %v, but it should be greater than 0", p.Name, p.Namespace, cpuUsage)
```

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
+				if !(cpuUsage > 0){
+					framework.Failf("Pods reported usage is %v, but it should be greater than 0", cpuUsage)
+				}
+				if !((.5*1.05) > cpuUsage){
```
```suggestion
				if cpuUsage >= .5*1.05 {
```

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
+					framework.Failf("Pods reported usage is %v, but it should be greater than 0", cpuUsage)
+				}
+				if !((.5*1.05) > cpuUsage){
+					framework.Failf("Pods reported usage is %v, but it should not exceed limit by >5%", cpuUsage)
```
```suggestion
					framework.Failf("Pod %s/%s reported usage is %v, but it should not exceed limit by >5%", p.Name, p.Namespace, cpuUsage)
```

pohly replied:
Please do the same changes also in the other file.

NikhilSharmaWe commented:
@pohly made changes according to your review.

pohly commented on test/e2e/windows/cpu_limits.go:
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
+					framework.Failf"Pod %s/%s reported usage is %v, but it should be greater than 0", p.Name, p.Namespace, cpuUsage)
+				}
+				if cpuUsage >= .5*1.05 {
+					framework.Failf("Pod %s/%s reported usage is %v, but it should not exceed limit by >5%", p.Name, p.Namespace, cpuUsage)
```
Wrong order of parameters.

pohly commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
-						framework.ExpectEqual(*podStats.Memory.WorkingSetBytes > 0, true, "Pod stats should not report 0 bytes for memory working set ")
-
+						if *podStats.CPU.UsageCoreNanoSeconds <= 0 {
+							framework.Failf("Pod stats report cpu usage equal to %v but it should be greater than 0", *podStats.CPU.UsageCoreNanoSeconds)
```
Include the pod namespace+name for which this fails.

pohly commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 						for _, containerStats := range podStats.Containers {
-							framework.ExpectEqual(containerStats.Logs != nil, true, "Pod stats should have container log stats")
+							if containerStats.Logs == nil {
+								framework.Failf("Pod stats should have container log stats, but it is equal to %v", nil)
```
Include the pod namespace+name and container name for which this fails.

NikhilSharmaWe replied:
Can you confirm that in this case Pod name and namespace will be podStats.PodRef.Name and podStats.PodRef.Namespace

pohly replied:
If those are valid fields, then yes. I don't know this struct.

pohly commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
+							}
 							framework.ExpectEqual(*containerStats.Logs.AvailableBytes > 0, true, "container log stats should not report 0 bytes for AvailableBytes")
+							if *containerStats.Logs.AvailableBytes <= 0 {
+								framework.Failf("container log stats report %v bytes for AvailableBytes, but it should be greater than 0", *containerStats.Logs.AvailableBytes)
```
Same here.

pohly commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -94,7 +102,9 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 				framework.Logf("Getting kubelet stats for node %v took an average of %v milliseconds over %v iterations", targetNode.Name, avgDurationMs, iterations)
-				framework.ExpectEqual(durationMatch, true, "Collecting kubelet stats should not take longer than 10 seconds")
+				if !durationMatch {
+					framework.Failf("Collecting kubelet stats take %v seconds but it should not take longer than 10 seconds", durationMatch)
```
s/take/took/

pohly commented on test/e2e/windows/kubelet_stats.go:
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 						for _, containerStats := range podStats.Containers {
-							framework.ExpectEqual(containerStats.Logs != nil, true, "Pod stats should have container log stats")
+							if containerStats.Logs == nil {
+								framework.Failf("Pod %s/%s stats should have container log stats, but it is equal to %v", podStats.PodRef.Namespace, podStats.PodRef.Name,nil)
```
There's no point in passing a nil as parameter. Just put that into the message and also make it shorts ("but it is nil").

NikhilSharmaWe commented:
@pohly could you please review the changes made.

NikhilSharmaWe commented:
/ok-to-test

NikhilSharmaWe commented:
@pohly I am unable to understand the cause error showing here 
`There are no nodes that your pod can schedule to - check your requests, tolerations, and node selectors (0/102 nodes are available: 16 Insufficient memory, 3 node(s) had taint {dedicated: greenhouse}, that the pod didn't tolerate, 99 Insufficient cpu.)`. The same is also showing in [PR](https://github.com/kubernetes/kubernetes/pull/106200) for files under autoscaling

pohly commented:
You need to fix the Go syntax.

NikhilSharmaWe commented:
@pohly Merged conflict appears when I try to rebase the branch.
`user@DESKTOP-ISNGT7P MINGW64 /e/githubRepos/kubernetes (betterOutputWindows)
$ git rebase upstream/master
error: could not apply 17fec5f1d6f... Changed code to improve output for test/e2e/windows
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 17fec5f1d6f... Changed code to improve output for test/e2e/windows
Auto-merging test/e2e/windows/kubelet_stats.go
CONFLICT (content): Merge conflict in test/e2e/windows/kubelet_stats.go`

pohly commented:
After the merge conflict you must manually edit `test/e2e/windows/kubelet_stats.go` to resolve the conflict.

NikhilSharmaWe commented:
@pohly Made another commit for resolving the commit, it should be fine now.
Just one Query is that if I try to convert the changes again in a single commit  then will the merge conflict will occur again or not, because it should be only 1 commit in a single PR.

pohly commented:
```
test/e2e/windows/cpu_limits.go:90:21: syntax error: unexpected literal "Pod %s/%s reported usage is %v, but it should be greater than 0" at end of statement
```

pohly commented:
There's a govet error.

NikhilSharmaWe commented:
@pohly could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.

NikhilSharmaWe commented:
@pohly by the way I changed 5% to 5 percent because it was showing govet error

pohly commented:
> by the way I changed 5% to 5 percent because it was showing govet error

You can escape `%` in a format string with `%%`.

> could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.

Sorry, I don't know which editor will work well in your environment. But you really need to find a way to find to avoid such simple mistakes like broken formatting or code that has syntax errors before submitting a PR update. Every such update kicks of all tests jobs, which costs the project real money and resources that could be spent better.

NikhilSharmaWe commented:
> > by the way I changed 5% to 5 percent because it was showing govet error
> 
> You can escape `%` in a format string with `%%`.
> 
> > could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.
> 
> Sorry, I don't know which editor will work well in your environment. But you really need to find a way to find to avoid such simple mistakes like broken formatting or code that has syntax errors before submitting a PR update. Every such update kicks of all tests jobs, which costs the project real money and resources that could be spent better.

I am so sorry about that, I will keep this in mind in my upcoming PR's

NikhilSharmaWe commented:
/kind cleanup

pohly reviewed (approved):
/lgtm

pohly commented:
/assign @ddebroy

For approval.

marosset commented:
/cc @jsturtevant

jsturtevant commented:
@NikhilSharmaWe thanks! Any reason not to include updates to the new metric non-serial test you added recently? 

https://github.com/kubernetes/kubernetes/blob/51b94de68f0889e4e9bd5bc7673041f1275ca175/test/e2e/windows/kubelet_stats.go#L149-L150

NikhilSharmaWe commented:
@jsturtevant I think I made the changes when the changes made for adding new metric were not merged. I will be updating the PR soon.

NikhilSharmaWe committed:
Changed code to improve output for test/e2e/windows

jsturtevant commented:
/lgtm
/approve
/milestone v1.23
